### PR TITLE
Move Twig Helpers from coreshop/pimcore to pimcore/pimcore

### PR DESCRIPTION
**Pull request by dpfaffenbauer**
These are quite some handy twig extensions that should be part of the Core.

@brusch there are many more things in coreshop/pimcore that might make sense, if you mgiht take a look at it what you think might help here as well: https://github.com/coreshop/CoreShop/tree/master/src/CoreShop/Component/Pimcore

### CODE REVIEW

File: lib/Twig/Extension/AssetHelperExtensions.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+use Twig\Extension\AbstractExtension;
+use Twig\TwigTest;
+
+final class AssetHelperExtensions extends AbstractExtension
+{
+    public function getTests(): array
+    {
+        return [
+            new TwigTest('pimcore_asset', static function ($object) {
+                return is_object($object) && $object instanceof Asset;
```
Comment (jdreesen) [inline lib/Twig/Extension/AssetHelperExtensions.php:30]: The `is_object` check should not be needed. 
The PHP docs say: 
`instanceof does not throw any error if the variable being tested is not an object, it simply returns FALSE. Constants, however, were not allowed prior to PHP 7.3.0.`

```suggestion
                return $object instanceof Asset;
```

Comment (dpfaffenbauer) [reply lib/Twig/Extension/AssetHelperExtensions.php:30]: true, not sure why I used that here, it for sure had a reason :) but might be to older PHP Versions, will remove it.

**Comment by dpfaffenbauer** [comment]
@jdreesen done

**Comment by brusch** [comment]
Thanks a lot for your PR! 
Regarding other Twig extensions ... we're quite open here, proposals are definitely welcome! 🙏😊

**Comment by brusch** [comment]
I'll leave this PR open until we have a complete set of helpers we gonna want to have in the core.

**Comment by dpfaffenbauer** [comment]
@brusch the question is what helpers ;). Some seem a bit outdated (like coreshop_path, since I think you already implemented something to alle path(data_object) right?)

**Comment by dpfaffenbauer** [comment]
@dvesh3 commited your changes

**Comment by dvesh3** [comment]
> @dvesh3 commited your changes

thanks! I guess we also have to change the type hints.

**Comment by dpfaffenbauer** [comment]
> > @dvesh3 commited your changes
> 
> thanks! I guess we also have to change the type hints.

Thinking about it, since it is twig helpers, I think we should not allow Thumnbail or array as a config. I am using this helpers for years now and never had the need to use anything else than string.

**Comment by dvesh3** [comment]
> > > @dvesh3 commited your changes
> > 
> > 
> > thanks! I guess we also have to change the type hints.
> 
> Thinking about it, since it is twig helpers, I think we should not allow Thumnbail or array as a config. I am using this helpers for years now and never had the need to use anything else than string.

 @dpfaffenbauer TBH not sure, if we really need these functions since you can call getThumbnail directly on image like,. 
```twig
{{ image.thumbnail('exampleScaleWidth').html({
    'class': 'thumbnail-class',
    'data-my-name': 'my value',
}) }}
```

**Comment by dpfaffenbauer** [comment]
@dvesh3 this is more of a separation of concerns. It bugs me that the model classes in Pimcore have too much responsability. The image itself should not be responsible for creating the thumbnail, a service should do that. this PR will make Pimcore future proof in case this ever gets refactored.

**Comment by dvesh3** [comment]
> @dvesh3 this is more of a separation of concerns. It bugs me that the model classes in Pimcore have too much responsability. The image itself should not be responsible for creating the thumbnail, a service should do that. this PR will make Pimcore future proof in case this ever gets refactored.

@dpfaffenbauer got your point 👍 let's revert my suggestions then 😉

File: lib/Twig/Extension/ImageThumbnailExtension.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+     * @param string $thumbnail
+     * @param array  $options
+     * @param array  $removeAttributes
+     * @param bool   $deferred
+     *
+     * @return string
+     */
+    public function getImageThumbnailHtml(
+        Image $image,
+        string|array|Image\Thumbnail\Config $thumbnail,
```
Comment (dvesh3) [inline lib/Twig/Extension/ImageThumbnailExtension.php:72]:
```suggestion
        string $thumbnail,
```

File: lib/Twig/Extension/ImageThumbnailExtension.php [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    {
+        return [
+            new TwigFunction('pimcore_image_thumbnail', [$this, 'getImageThumbnail'], ['is_safe' => ['html']]),
+            new TwigFunction('pimcore_image_thumbnail_html', [$this, 'getImageThumbnailHtml'], ['is_safe' => ['html']]),
+        ];
+    }
+
+    /**
+     * @param Image  $image
+     * @param string|array|Image\Thumbnail\Config $thumbnail
```
Comment (dvesh3) [inline lib/Twig/Extension/ImageThumbnailExtension.php:51]:
```suggestion
     * @param string $thumbnail
```

**Comment by dpfaffenbauer** [comment]
🎉

**Comment by dvesh3** [comment]
@dpfaffenbauer thanks a lot 🙏

**Comment by brusch** [comment]
Sorry, one more, thing  .... docs 😉
Can someone please add them here: 
https://github.com/pimcore/pimcore/blob/34896ec5737a49cd5ea51898391fefb441c947d2/doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md#L115

Please also check if there are some others which need to be added. 
Thanks.

**Pull request merged**